PYTHON-5172 Remove __init__ from BinaryVector dataclass

caseyclements opened the pull request:
This is an alternative approach to #2162.

blink1073 commented:
```python
ImportError while loading conftest '/home/runner/work/mongo-python-driver/mongo-python-driver/test/conftest.py'.
test/__init__.py:35: in <module>
    from pymongo.uri_parser import parse_uri
pymongo/__init__.py:91: in <module>
    from pymongo.asynchronous.mongo_client import AsyncMongoClient
pymongo/asynchronous/mongo_client.py:61: in <module>
    from bson.codec_options import DEFAULT_CODEC_OPTIONS, CodecOptions, TypeRegistry
bson/__init__.py:105: in <module>
    from bson.binary import (
bson/binary.py:231: in <module>
    class BinaryVector:
E   ValueError: 'padding' in __slots__ conflicts with class variable
```

caseyclements commented:
> ```python
> ImportError while loading conftest '/home/runner/work/mongo-python-driver/mongo-python-driver/test/conftest.py'.
> test/__init__.py:35: in <module>
>     from pymongo.uri_parser import parse_uri
> pymongo/__init__.py:91: in <module>
>     from pymongo.asynchronous.mongo_client import AsyncMongoClient
> pymongo/asynchronous/mongo_client.py:61: in <module>
>     from bson.codec_options import DEFAULT_CODEC_OPTIONS, CodecOptions, TypeRegistry
> bson/__init__.py:105: in <module>
>     from bson.binary import (
> bson/binary.py:231: in <module>
>     class BinaryVector:
> E   ValueError: 'padding' in __slots__ conflicts with class variable
> ```

Yeah. This is why I this up. Thought maybe you or @ShaneHarvey  could see the issue. In #2162, I've removed the dataclass decorator. I've already implemented __init__, __repr__, and don't want ==, >=.... What do you guys think?

blink1073 commented:
Ah, that explains why we added `__init__` in the first place.  I think we have no choice since the slots feature isn't available until Python 3.10.

ShaneHarvey commented:
I see, then let's close this PR and continue with the other one.
